Content.Shared/GameObjects/Components/Storage/SharedItemComponent.cs

```diff
@@ -0,0 +1,227 @@
+        }
+        [DataField("sprite")]
+        private string? _rsiPath;
+
+        public override ComponentState GetComponentState(ICommonSession player)
+        {
+            return new ItemComponentState(Size, EquippedPrefix, Color, RsiPath);
+        }
+
+        public override void HandleComponentState(ComponentState? curState, ComponentState? nextState)
+        {
+            base.HandleComponentState(curState, nextState);
+
+            if (curState == null)
+                return;
+
+            if (curState is not ItemComponentState state)
+                return;
+
+            Size = state.Size;
+            EquippedPrefix = state.EquippedPrefix;
+            Color = state.Color;
+            RsiPath = state.RsiPath;
+        }
+
+        /// <summary>
+        ///     If a player can pick up this item.
+        /// </summary>
+        public bool CanPickup(IEntity user)
+        {
+            if (!ActionBlockerSystem.CanPickup(user))
+                return false;
+
+            if (user.Transform.MapID != Owner.Transform.MapID)
+                return false;
+
+            if (!Owner.TryGetComponent(out IPhysBody? physics) || physics.BodyType == BodyType.Static)
+                return false;
+
+            return user.InRangeUnobstructed(Owner, ignoreInsideBlocker: true, popup: true);
+        }
+
+        void IEquipped.Equipped(EquippedEventArgs eventArgs)
+        {
+            EquippedToSlot();
+        }
+
+        void IUnequipped.Unequipped(UnequippedEventArgs eventArgs)
+        {
+            RemovedFromSlot();
+        }
+
+        void IExAct.OnExplosion(ExplosionEventArgs eventArgs)
+        {
+            var source = eventArgs.Source;
+            var target = eventArgs.Target.Transform.Coordinates;
+            var dirVec = (target.ToMapPos(Owner.EntityManager) - source.ToMapPos(Owner.EntityManager)).Normalized;
+
+            var throwForce = eventArgs.Severity switch
+            {
+                ExplosionSeverity.Destruction => 3.0f,
+                ExplosionSeverity.Heavy => 2.0f,
+                _ => 1.0f,
+            };
+            ThrowItem(dirVec * throwForce);
+        }
+
+        bool IInteractHand.InteractHand(InteractHandEventArgs eventArgs)
+        {
+            return TryPutInHand(eventArgs.User);
+        }
+
+        void IThrown.Thrown(ThrownEventArgs eventArgs)
+        {
+            if (!Owner.TryGetComponent(out PhysicsComponent? physicsComponent)) return;
+
+            foreach (var fixture in physicsComponent.Fixtures)
+            {
+                fixture.CollisionLayer |= (int) CollisionGroup.MobImpassable;
+            }
+        }
+
+        void ILand.Land(LandEventArgs eventArgs)
+        {
+            if (!Owner.TryGetComponent(out PhysicsComponent? physicsComponent)) return;
+
+            foreach (var fixture in physicsComponent.Fixtures)
+            {
+                fixture.CollisionLayer &= ~(int) CollisionGroup.MobImpassable;
+            }
+        }
```

Comment:
This will need to be fixtures (I think I already do it under ThrownItem so can probably delete).